lib/util/file_cache.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Burrito.Util.FileCache do
+  require Logger
+
+  @cache_namespace "burrito_file_cache"
+
+  @spec init_local_cache :: :ok
+  def init_local_cache do
+    get_cache_dir() |> File.mkdir_p!()
+  end
+
+  @spec fetch(binary()) :: {:hit, binary} | {:miss, nil}
+  def fetch(key) when is_binary(key) do
+    cache_dir = get_cache_dir()
+    full_path = Path.join(cache_dir, [key])
+
+    if File.exists?(full_path) do
+      {:hit, File.read!(full_path)}
+    else
+      {:miss, nil}
+    end
```

Comment:
nitpick: to avoid hitting the fs twice you may use:

```suggestion
    case File.read(full_path) do
      {:ok, data} -> {:hit, data}
      {:error, :enoent} -> :miss
    end
```